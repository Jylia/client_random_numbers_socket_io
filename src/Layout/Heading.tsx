import React, { Component } from 'react'
import {
    Button,
    Container,
    Header,
    Icon,
} from 'semantic-ui-react';

interface IHeadingProps {
    isMobile: boolean;
}

interface IHeadingState {}

class Heading extends Component<IHeadingProps, IHeadingState> {

    render () {
        return (
            <Container text>
            <Header
                as='h1'
                content='Random number charts'
                inverted
                style={{
                    fontSize: this.props.isMobile ? '2em' : '4em',
                    fontWeight: 'normal',
                    marginBottom: 0,
                    marginTop: this.props.isMobile ? '1.5em' : '3em',
                }}
            />
            <Header
                as='h2'
                content='Look at changes in real-time'
                inverted
                style={{
                    fontSize: this.props.isMobile ? '1.5em' : '1.7em',
                    fontWeight: 'normal',
                    marginTop: this.props.isMobile ? '0.5em' : '1.5em',
                }}
            />
            <Button primary size='huge'>
                Get Started
                <Icon name='arrow right' />
            </Button>
        </Container>
        );
    }
}

export default Heading;