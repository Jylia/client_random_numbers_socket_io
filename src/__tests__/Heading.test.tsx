import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Heading from '../Layout/Heading';

it('renders without crashing for mobile', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Heading isMobile={true} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing for desktop', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Heading isMobile={false} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly  for mobile', () => {
  const tree = renderer
      .create(<Heading isMobile={true} />)
      .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly  for desktop', () => {
  const tree = renderer
      .create(<Heading isMobile={false} />)
      .toJSON();
  expect(tree).toMatchSnapshot();
});
