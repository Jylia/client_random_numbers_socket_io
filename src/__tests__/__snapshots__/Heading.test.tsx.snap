// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  for desktop 1`] = `null`;

exports[`renders correctly  for mobile 1`] = `null`;
